datafusion/functions-aggregate/src/array_agg.rs

```diff
@@ -288,10 +300,23 @@ impl Accumulator for ArrayAggAccumulator {
             return internal_err!("expects single batch");
         }
 
-        let val = Arc::clone(&values[0]);
+        let val = &values[0];
+        let nulls = if self.ignore_nulls {
+            val.logical_nulls()
+        } else {
+            None
+        };
+
+        let val = match nulls {
+            Some(nulls) if nulls.null_count() >= val.len() => return Ok(()),
```

Comment:
how could the null count be greater than the length? or is the `>=` a defensive coding mechanism?

Reply:
It shouldn't be - I guess we can use `==` but does it make a difference?